backend/src/main/java/com/huellapositiva/application/controller/RecoveryPasswordApiController.java

```diff
@@ -0,0 +1,85 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+
+@Controller
+@RequestMapping("/api/v1/restore-password")
+public class RecoveryPasswordApiController {
+
+    @Autowired
+    FetchCredentialsAction credentialsAction;
+
+    @Operation(
+            summary = "Send an email to recovery password",
+            description = "Fetch the credential, create an email with a link to recover password, send the email.",
+            tags = {"contactPerson, volunteers, recoveryPassword"}
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "204",
+                            description = "No content, email sent successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "404",
+                            description = "Not found, email not found in database."
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @GetMapping("/sendRecoveryPasswordEmail/{email}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void sendEmailRecovery(@PathVariable String email){
+        try {
+            credentialsAction.executeGenerationRecoveryPasswordEmail(email);
+        } catch (UserNotFoundException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Email not found");
+        }
+    }
+
+    @Operation(
+            summary = "Update the password",
+            description = "Receive the new password, update it in the database and send a confirmation email .",
+            tags = {"contactPerson, volunteers, recoveryPassword"}
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "204",
+                            description = "No content, password updated and confirmation email sent successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "423",
+                            description = "Locked, can not access to the resource because the time has expired."
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @PostMapping("/changePassword")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void changePassword(@RequestParam("hash") String hash, @RequestParam("newPassword") String password){
```

Comment:
We should not use request parameters for this purpose. We are implementing a JSON API and we should favor this approach unless inevitable.

The exception could be the hash, that could be considered an id of a resource and therefore be part of the URI

`POST /changePassword/{hash}
{
  "newPassword": "your-new-password"
}`

Reply:
Ok, Ibai, we fixed it.